components/exercises/chapter-5-oscar/Flattening/Flattening.tsx

```diff
@@ -0,0 +1,67 @@
+import { getFlattenedArray } from './getFlattenedArray';
+
+const arrays = [
+  ['❤️✨', '🚀🌟', '💭💪'],
+  ['🌅🔄', '💪😃'],
+  ['🚫🙅‍♂️', '💪🏆'],
+];
+
+export const phrases = [
+  { icon: '❤️✨', phrase: 'Believe in yourself, and you are halfway there.' },
+  { icon: '🚀🌟', phrase: 'Success is a journey, not a destination.' },
+  { icon: '💭💪', phrase: 'Dream big and dare to fail.' },
+  { icon: '🌅🔄', phrase: 'Every setback is a setup for a comeback.' },
+  { icon: '💪😃', phrase: 'You are stronger than you think.' },
+  { icon: '🚫🙅‍♂️', phrase: 'The only limit is the one you set for yourself.' },
+  {
+    icon: '💪🏆',
+    phrase: 'Hard work beats talent when talent does not work hard.',
+  },
+];
+
+/**
+ * Renders a component that displays flattened phrases from an array of arrays.
+ * Each phrase is represented by a JSX element containing an icon and a corresponding phrase.
+ */
+export default function Flattening(): JSX.Element {
+  /**
+   * Retrieves the flattened array of phrases from the given array of arrays, and maps each phrase to a JSX element containing the corresponding icon and phrase.
+   * @param arrays - The array of arrays to be flattened.
+   * @returns An array of JSX elements representing the phrases for the flattened icons. 
+   */
+  const getPhrase = getFlattenedArray(arrays).map((icon, index) => {
+    const matchingIcon = phrases.find((phrase) => phrase.icon === icon);
```

Comment:
this shouldn't be named `matchingIcon` either because the element this constant binds to is not the icon, but the object that contains the icon + phrase

Reply:
refactored to matchingIconWithPhrase